src/main/java/com/google/jenkins/plugins/storage/StdoutUploadStep.java

```diff
@@ -0,0 +1,186 @@
+    upload.setSharedPublicly(sharedPublicly);
+  }
+
+  public boolean isSharedPublicly() {
+    return upload.isSharedPublicly();
+  }
+
+  /**
+   * Whether to indicate in metadata that the file should be viewable inline in web browsers, rather
+   * than requiring it to be downloaded first.
+   */
+  @DataBoundSetter
+  public void setShowInline(boolean showInline) {
+    upload.setShowInline(showInline);
+  }
+
+  public boolean isShowInline() {
+    return upload.isShowInline();
+  }
+
+  /**
+   * The path prefix that will be stripped from uploaded files. May be null if no path prefix needs
+   * to be stripped.
+   *
+   * <p>Filenames that do not start with this prefix will not be modified. Trailing slash is
+   * automatically added if it is missing.
+   */
+  @DataBoundSetter
+  public void setPathPrefix(@Nullable String pathPrefix) {
+    upload.setPathPrefix(pathPrefix);
+  }
+
+  @Nullable
+  public String getPathPrefix() {
+    return upload.getPathPrefix();
+  }
+
+  public String getLogName() {
+    return upload.getLogName();
+  }
+
+  public String getBucket() {
+    return upload.getBucket();
+  }
+
+  /** The unique ID for the credentials we are using to authenticate with GCS. */
+  public String getCredentialsId() {
+    return credentialsId;
+  }
+
+  private final String credentialsId;
+
+  @Override
+  public BuildStepMonitor getRequiredMonitorService() {
+    return BuildStepMonitor.NONE;
+  }
+
+  @Override
+  public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener)
+      throws IOException {
+    // setForFailedJobs was set to true in the constructor. However,
+    // some jobs might have been created before that bug fix.
+    // For those, set this here as well.
+    upload.setForFailedJobs(true);
+
+    try {
+      upload.perform(getCredentialsId(), run, workspace, listener);
+    } catch (UploadException e) {
+      throw new IOException("Could not perform upload", e);
+    }
+  }
+
+  /** Descriptor for the class. */
+  @Extension
+  @Symbol("googleStorageUpload")
+  public static class DescriptorImpl extends BuildStepDescriptor<Builder> {
+
+    /** {@inheritDoc} */
+    @Override
+    public String getDisplayName() {
+      return Messages.StdoutUpload_BuildStepDisplayName();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean isApplicable(Class<? extends AbstractProject> jobType) {
+      return true;
+    }
+
+    @Override
+    public Builder newInstance(StaplerRequest req, JSONObject formData) throws FormException {
+      // Since the config form lists the optional parameter pathPrefix as
+      // inline, it will be passed through even if stripPathPrefix is false.
+      // This might cause problems if the user, for example, fills in the field
+      // and then unchecks the checkbox. So, explicitly remove pathPrefix
+      // whenever stripPathPrefix is false.
+      if (Boolean.FALSE.equals(formData.remove("stripPathPrefix"))) {
```

Comment:
!formData.remove("stripPathPrefix")

Reply:
Remove returns an Object
```
   /**
    * Remove a name and its value, if present.
    *
    * @param key The name to be removed.
    * @return The value that was associated with the name, or null if there was
    *         no value.
    */
   public Object remove( String key ) {
      verifyIsNull();
      return this.properties.remove( key );
   }
```
Not sure of best way to make this check for false? Using ! doesn't typecheck. Would casting to a Boolean be better?

Reply:
oic, didn't realize. Thanks for clarifying :)